### PR TITLE
Add titan_defconfig

### PR DESCRIPTION
**Pull request by luca020400**
With every falcon change

### CODE REVIEW

File: arch/arm/configs/titan_defconfig [hunk 1]
```diff
@@ -43,7 +69,7 @@ CONFIG_IRQ_WORK=y
 CONFIG_EXPERIMENTAL=y
 CONFIG_INIT_ENV_ARG_LIMIT=32
 CONFIG_CROSS_COMPILE=""
-CONFIG_LOCALVERSION=""
+CONFIG_LOCALVERSION="-Optimus-Titan-LP"
```
Comment (luca020400) [inline arch/arm/configs/titan_defconfig]: Need to be changed for release

Comment (desaishivam26) [inline arch/arm/configs/titan_defconfig]: Thanks for that!!
I was about to do it

File: arch/arm/configs/titan_defconfig [hunk 2]
```diff
@@ -229,12 +255,13 @@ CONFIG_IOSCHED_NOOP=y
...
 CONFIG_IOSCHED_DEADLINE=y
 CONFIG_IOSCHED_ROW=y
+CONFIG_IOSCHED_FIOPS=y
 CONFIG_IOSCHED_CFQ=y
 # CONFIG_DEFAULT_DEADLINE is not set
 # CONFIG_DEFAULT_ROW is not set
 # CONFIG_DEFAULT_CFQ is not set
-CONFIG_DEFAULT_NOOP=y
-CONFIG_DEFAULT_IOSCHED="noop"
+# CONFIG_DEFAULT_NOOP is not set
```
Comment (luca020400) [inline arch/arm/configs/titan_defconfig]: Should we add CONFIG_DEFAULT_FIOPS ?

File: arch/arm/configs/titan_defconfig [hunk 3]
```diff
@@ -17,7 +43,7 @@ CONFIG_TRACE_IRQFLAGS_SUPPORT=y
 CONFIG_HARDIRQS_SW_RESEND=y
 CONFIG_GENERIC_IRQ_PROBE=y
 CONFIG_ARM_TICKET_LOCKS=y
-CONFIG_RWSEM_GENERIC_SPINLOCK=y
+#CONFIG_RWSEM_GENERIC_SPINLOCK is not set
```
Comment (luca020400) [inline arch/arm/configs/titan_defconfig]: Why this got disabled ?

Comment (desaishivam26) [inline arch/arm/configs/titan_defconfig]: It's disabled for this. It makes use of optimized RWSEM algorithm
CONFIG_RWSEM_XCHGADD_ALGORITHM=y

File: arch/arm/configs/titan_defconfig (hunk 2 above)
Comment (desaishivam26) [inline arch/arm/configs/titan_defconfig]: Why not?

Comment (luca020400) [inline arch/arm/configs/titan_defconfig]: Done

File: arch/arm/configs/titan_defconfig (hunk 3 above)
Comment (luca020400) [inline arch/arm/configs/titan_defconfig]: Ok thanks
Ready to merge

Comment (desaishivam26) [inline arch/arm/configs/titan_defconfig]: Hey can u create a pull request to my Optimus-Titan branch?
The one that u created is to falcon branch. And yes if possible dont include kexec change in defconfig bcoz its already done

**Comment by luca020400** [comment]
What happened ? xD

**Comment by desaishivam26** [comment]
You created a pull request to falcon branch. I have a separate Titan branch. Pls create a pull request to that.(without kexec change)

**Comment by luca020400** [comment]
Why two branches ?
Both can live in the same branch ;)
Less work
And kexec works fine on titan

**Comment by desaishivam26** [comment]
I have noticed that there are some titan specific commits, so its good to have a separate branch.I cherry picked kexec commit and it already has the defconfig change. so i am asking u to do it.

**Comment by luca020400** [comment]
I'm trying to say that every falcon commit works on titan too
Titan Kexec = Falcon Kexec
Falcon Faux Sound = Titan Faux Sound

**Comment by desaishivam26** [comment]
Lol I asked Anas Karbila about it. He said its better to have a separate branch just to be on a safer side

**Comment by luca020400** [comment]
Nah :p

**Comment by desaishivam26** [comment]
If u could create a pull request? Or i will have to do all again xD

**Pull request merged**